base/abstractarray.jl

```diff
@@ -37,7 +37,7 @@ Base.OneTo(6)
 """
 function indices(A::AbstractArray{T,N}, d) where {T,N}
     @_inline_meta
-    d <= N ? indices(A)[d] : OneTo(1)
+    d <= N ? indices(A)[d] : one(indicestype(typeof(A),d))
```

Comment:
Personally, I would rather prefer that functions in base like `broadcast` never needed to call this function in this (second/negative) case... I.e. If custom types neglect to implement indexing operations other than linear or exactly `N` dimensional Cartesian, then it would be great if everything respected that and just worked (this surprised me in some point in the past).